cogs/general.py

```diff
@@ -210,30 +224,80 @@ async def changecolor(self, context, color):
+                            
+                        topRole = userRoles[-1]
+                        await topRole.edit(colour=nextcord.Colour(int(randomColor.replace("#", ""), 16)))   # changes top role of user
+
+                        newEmbed = nextcord.Embed ( # create new embed
+                            title="Success!",
+                            description="Color has been changed! The hex value is: " + randomColor + ". The contrast ratio is: " + str(round(contrast, 4)) + ":1",
+                            color=int(randomColor.replace("#", ""), 16)
+                        )
+                        await interaction.message.edit(embed=newEmbed)  # modify existing embed
+                
+            else:   # if message value after !changecolor is not "random"
+                contrast = self.contrast("#36393f", color)
+                chosenColor = color
+                if contrast < limit:
+                    embed = nextcord.Embed(
+                        title="Error",
+                        description="Color does not have enough contrast. That color has a contrast ratio of: " + str(round(contrast, 4)) + ":1. It needs to be above " + limit + ":1.",
+                        color=int(color.replace("#", ""), 16)
+                    )
+                    await context.send(embed=embed)
+                    return
+                    
             if len(userRoles) > 1:
                 topRole = userRoles[-1]
-                await topRole.edit(colour=nextcord.Colour(int(color.replace("#", ""), 16)))
+                await topRole.edit(colour=nextcord.Colour(int(chosenColor.replace("#", ""), 16)))
+
                 embed = nextcord.Embed(
                     title="Success!",
-                    description="Color has been changed! The contrast it has is " + str(round(contrast, 4)) + ":1",
-                    color=int(color.replace("#", ""), 16)
+                    description="Color has been changed! The hex value is: " + chosenColor + ". The contrast ratio is: " + str(round(contrast, 4)) + ":1",
+                    color=int(chosenColor.replace("#", ""), 16)
+                )
+
+                if color.lower() == "random":
+                    color_button.callback = color_callback
+                    view = View(timeout=1000)
+                    view.add_item(color_button)
+                    await context.send(embed=embed, view=view)
+                else:
+                    await context.send(embed=embed)
+            else:
+                embed = nextcord.Embed(
+                    title = "Error!",
+                    description = "The user has only one role. Color cannot be changed with this command.",
+                    color = config["error"]
                 )
                 await context.send(embed=embed)
-        except:
+                    
+        except Exception as e:  # generic error in discord, printed error in console
+            print(e)

```

Comment:
I think we want to leave this print out for production so it doesn't print an error to the console every time someone does a bad thing with the command